packages/rest/src/parse-json.ts

```diff
@@ -16,11 +16,35 @@
 /**
  * Factory to create a reviver function for `JSON.parse` to sanitize keys
  * @param reviver - Reviver function
+ * @param prohibitedKeys - An array of keys to be rejected
  */
-export function sanitizeJsonParse(reviver?: (key: any, value: any) => any) {
+export function sanitizeJsonParse(
+  reviver?: (key: any, value: any) => any,
+  prohibitedKeys?: string[],
+) {
+  prohibitedKeys = [
+    '__proto__',
+    'constructor.prototype',
+    ...(prohibitedKeys ?? []),
+  ];
   return (key: string, value: any) => {
-    if (key === '__proto__')
-      throw new Error('JSON string cannot contain "__proto__" key.');
+    if (key === '__proto__') {
+      // Reject `__proto__`
+      throw new Error(`JSON string cannot contain "${key}" key.`);
+    }
+    if (
+      key === 'constructor' &&
+      value != null &&
+      Object.prototype.hasOwnProperty.call(value, 'prototype')
```

Comment:
should we check the value's field contains `prototype` as well? like:

```ts
const text = '{"x": "1", "y": {"constructor": {"prototype.z": {"foo": 2}}}}';
```

Reply:
IIUC, only `{"constructor": {"prototype": ...}}` is dangerous.

Reply:
> IIUC, only `{"constructor": {"prototype": ...}}` is dangerous.

Why is that dangerous? If you could clarify. thanks :)

Reply:
See https://medium.com/node-modules/what-is-prototype-pollution-and-why-is-it-such-a-big-deal-2dd8d89a93c

Reply:
> See https://medium.com/node-modules/what-is-prototype-pollution-and-why-is-it-such-a-big-deal-2dd8d89a93c

Yikes. The fact that it pollutes all objects after one string is parsed...wow. Thanks for the link.